src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1220,41 +1226,48 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
                     }
 
                     explicitInterfaceImplementations = explicitInterfaceImplementationsBuilder.ToImmutableAndFree();
+
+                    MethodSymbol uniqueClassOverride = null;
+                    foreach (MethodSymbol method in explicitlyOverriddenMethods)
+                    {
+                        if (method.ContainingType.IsClassType())
+                        {
+                            if (uniqueClassOverride is { })
+                            {
+                                // not unique
+                                uniqueClassOverride = null;
+                                break;
+                            }
+
+                            uniqueClassOverride = method;
+                        }
+                    }
+
+                    if (uniqueClassOverride is { })
+                    {
+                        hasUniqueClassOverride = true;
+                        Interlocked.CompareExchange(ref AccessUncommonFields()._lazyExplicitClassOverride, uniqueClassOverride, null);
+                    }
                 }
 
+                // CONSIDER: what we'd really like to do is set this bit only in cases where the explicitly
+                // overridden method matches the method that will be returned by MethodSymbol.OverriddenMethod.
+                // Unfortunately, this MethodSymbol will not be sufficiently constructed (need IsOverride and MethodKind,
+                // which depend on this property) to determine which method OverriddenMethod will return.
+                _packedFlags.InitializeIsExplicitOverride(isExplicitFinalizerOverride: sawObjectFinalize, isExplicitClassOverride: hasUniqueClassOverride);
```

Comment:
>hasUniqueClassOverride [](start = 131, length = 22)

It looks like there  is a behavior change here, which is going to affect the value returned by IsExplicitClassOverride property. Could you please elaborate why is this necessary with respect to existing usage of this property and point me to the tests that observe and justify the behavior change around existing usage of the property.

Reply:
If by "existing usages" you mean prior to the addition of the covariant returns feature, the only usage is in `PEMethodSymbol.IsOverride`. This scenario only arises from metadata that doesn't have a mapping to C#. In such cases of metadata that doesn't map to the language we have the freedom to decide how to map it to the language APIs. We previously ignored the overrides in PE when deciding what (if any) to report as the overridden method. As a consequence there could have been a mismatch between `IsOverride` and `OverriddenMember` (true and null). Now we do not treat it as an override of any particular at all (except by the usual rules in the absence of the .override) if there is no unique override, these two properties align, and we do pay attention to the override in metadata if it is unique. The behavior is tested in `PEMethodSymbol_ExplicitlyOverriddenClassMethod_WhenAmbiguous` among other places.

If and when we decide to support methods that have more than one language override, we will have to change a number of things, including changing "overridden symbol" to "overridden symbolS" (i.e. make it plural). There is no reason to do that today as such scenarios do not map to the language.

---
In reply to: [441975630](https://github.com/dotnet/roslyn/pull/44025#discussion_r441975630) [](ancestors = 441975630)

Reply:
I think this behavior change should be reverted. I believe it is not related to the feature and unnecessary. I will not sign off like that. If you want to make a change like that, follow the standard route: open an issue, describe what the problem is, get support of the team, etc.

---
In reply to: [442308377](https://github.com/dotnet/roslyn/pull/44025#discussion_r442308377) [](ancestors = 442308377,441975630)

Reply:
I will revert the behavior change to the method `IsOverride` and leave the API's inconsistent results to be dealt with at some future time.

---
In reply to: [442323736](https://github.com/dotnet/roslyn/pull/44025#discussion_r442323736) [](ancestors = 442323736,442308377,441975630)